mac/Keyman4MacIM/Keyman4MacIM/KMInputMethodAppDelegate.m

```diff
@@ -73,21 +73,47 @@ - (id)init {
                                                          forEventClass:kInternetEventClass
                                                             andEventID:kAEGetURL];
         
-        CFMachPortRef eventTap = CGEventTapCreate(kCGAnnotatedSessionEventTap, kCGHeadInsertEventTap, kCGEventTapOptionListenOnly, NSFlagsChangedMask, (CGEventTapCallBack)eventTapFunction, nil);
+        CFMachPortRef lowLevelEventTap = CGEventTapCreate(kCGAnnotatedSessionEventTap, kCGHeadInsertEventTap, kCGEventTapOptionListenOnly, NSFlagsChangedMask | NSLeftMouseDown | NSLeftMouseUp | NSOtherMouseDown | NSOtherMouseUp, (CGEventTapCallBack)eventTapFunction, nil);
```

Comment:
Can context change with `NSRightMouseDown` and `NSRightMouseUp`?

Reply:
Not sure. I'll need to test this with an external mouse. Using the keypad, Control-clicking the touchpad still registers in the low-level event as a left-mouse click.

Reply:
I think I read Apple sets the default right-click to be the same as left-click, but somewhere in Settings you can change that. Try for a 3-button mouse to test left, other, and right buttons respectively.

Reply:
It looks like when a context menu appears (which is typically what happens when the right mouse button is clicked), the system registers that as a new "client", so we automatically clear the context in the Keyman IM. So I don't think there's any need to trap right-click events.

Reply:
I did not see a place in settings to specify that the two buttons should behave the same, but I did see where I could switch which one is primary. So if I make the right mouse button primary, then you're correct: I can use the right mouse button to make a context change that Keyman does not properly detect. I'll rectify the code...

Reply:
Nevermind, I just realized I was testing with the wrong version of Keyman (master instead of the branch for this PR). It turns out that when you change the settings to flip which button is primary, it happens at a low enough level that the actual low-level event itself reflects the switch, so a left-click is reported as a right-click and vice versa. No code change needed.

Reply:
Wow! I just looked at the code and realized that your comment was actually questioning the need for tapping into the right and other events. (I forgot I had done that.)